x-pack/plugins/apm/server/lib/traces/get_trace.js

```diff
@@ -0,0 +1,49 @@
+/*
+* Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+* or more contributor license agreements. Licensed under the Elastic License;
+* you may not use this file except in compliance with the Elastic License.
+*/
+
+import { TRACE_ID } from '../../../common/constants';
+
+export async function getTrace({ traceId, setup }) {
+  const { start, end, client, config } = setup;
+
+  const params = {
+    index: config.get('apm_oss.transactionIndices'),
+    body: {
+      size: 1000,
+      query: {
+        bool: {
+          filter: [
+            { term: { [TRACE_ID]: traceId } },
+            {
+              range: {
```

Comment:
Just to clarify for myself: the trace ID would be enough to get the right data but adding the time range helps speed up the query, right?

Reply:
That is the hypothesis, yes. @simitt is helping me getting this benchmarked - we just need to get around to doing it :)

Reply:
And yes, if it turns out there are not perf improvements, we can drop time time range.